dask/async.py

```diff
@@ -373,7 +376,9 @@ def default_get_id():
 
 def get_async(apply_async, num_workers, dsk, result, cache=None,
               queue=None, get_id=default_get_id, raise_on_exception=False,
```

Comment:
Is this Queue now *always* a `queues.Queue`?  

If so then we should probably create it within the function rather than pass it in.